esmvalcore/cmor/_fixes/cmip6/fgoals_f3_l.py

```diff
@@ -0,0 +1,41 @@
+"""Fixes for CMIP6 FGOALS-f3-L."""
+import cftime
+import numpy as np
+
+from ..fix import Fix
+
+
+class AllVars(Fix):
+    """Fixes for all vars."""
+    def fix_metadata(self, cubes):
+        """Fix parent time units.
+
+        FGOALS-f3-L Amon data may have a bad time bounds spanning 20 days.
+
+        Parameters
+        ----------
+        cubes : iris.cube.CubeList
+            Input cubes.
+
+        Returns
+        -------
+        iris.cube.CubeList
+        """
+        for cube in cubes:
+            if cube.attributes['table_id'] == 'Amon':
+                time = cube.coord('time')
+                if cube.attributes['table_id'] == 'Amon' and \
+                        not (time.bounds[0, 1] == time.bounds[1, 0]):
+                    times = time.units.num2date(time.points)
+                    starts = [
+                        cftime.DatetimeNoLeap(c.year, c.month, 1)
+                        for c in times
+                    ]
+                    ends = [
+                        cftime.DatetimeNoLeap(c.year, c.month +
+                                              1, 1) if c.month < 12 else
+                        cftime.DatetimeNoLeap(c.year + 1, 1, 1) for c in times
+                    ]
+                    time.bounds = time.units.date2num(
+                        np.stack([starts, ends], -1))
```

Comment:
is the fault discoverable immediately by looking at the first and second `[b1, b2]` bounds? Or could be somewhere deeper in the bounds array? I also think it's prob best if you just say `time.bounds = None; time.guess_bounds()` - bit more general?

Reply:
From what I have seen it's concerning all time bounds. I added a check over all time periods though.

Concerning boundary guessing, I don't consider it the way forward, as it simple finds bounds in the middle between the two points, which is not correct in the case of monthly data. Leading to e.g. DimCoord([1850-01-16 12:00:00, 1850-02-15 00:00:00, 1850-03-16 12:00:00 ...] , bounds=[[1850-01-01 18:00:00, 1850-01-31 06:00:00], [1850-01-31 06:00:00, 1850-03-01 18:00:00], [1850-03-01 18:00:00, 1850-03-31 18:00:00], ...]

Reply:
OK - am not a big fan of guessing bounds myself but it may be easier if the result is the same as the coded by hand one, which is not the case here :+1: